core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -288,12 +338,13 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                     }
                 }
 
-                if (msg instanceof LastHttpContent) {
+                // If a WebSocket session established, ignore the LastHttpContent.
+                if (!webSocketUpgradeContext.webSocketSessionEstablished() &&
+                    msg instanceof LastHttpContent) {
```

Comment:
The existing `Http1RequestDecoder` behaves based on the type of `DecodedHttpRequest` and the type of `msg`. `webSocketUpgradeContext` seems not to be mixed with them in the same context. What do you think of introducing `WebsocketDecodedHttpRequest` to more seamlessly handle WebSocket requests in `Http1RequestDecoder`?